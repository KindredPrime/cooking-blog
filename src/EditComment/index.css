.EditComment__submit {
  margin-right: 10px;
}