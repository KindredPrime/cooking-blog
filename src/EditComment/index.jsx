import { Component } from 'react';
import PropTypes from 'prop-types';
import APIError from '../APIError/index';
import './index.css';

class EditComment extends Component {
  state = {
    content: '',
    error: null
  };

  handleSubmit(e) {
    e.preventDefault();

    this.props.handleSubmit(this.props.id, this.state.content)
      .then(() => {
        this.setState({
          error: null
        });
      })
      .catch((error) => {
        this.setState({
          error
        })
      });
  }

  handleCancel(e) {
    e.preventDefault();

    this.props.handleCancel();
  }

  componentDidMount() {
    this.setState({
      content: this.props.initialContent
    });
  }

  render() {
    const { error } = this.state;

    return (
      <div className="EditComment">
        <form className="EditComment" onSubmit={(e) => this.handleSubmit(e)}>
          <textarea
            id="content"
            name="content"
            defaultValue={this.props.initialContent}
            onChange={(e) => this.setState({
              content: e.target.value
            })}
            required
          ></textarea>

          <div className="EditComment__button-wrapper">
            <button type="submit" className="EditComment__submit">
              Submit
            </button>

            <button
              type="submit"
              onClick={(e) => this.handleCancel(e)}
            >
              Cancel
            </button>
          </div>
        </form>

        {error && <APIError message={error.message} />}
      </div>
    );
  }
}

EditComment.propTypes = {
  initialContent: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleCancel: PropTypes.func.isRequired
};

export default EditComment;