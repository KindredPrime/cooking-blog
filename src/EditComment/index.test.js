import ReactDOM from 'react-dom';
import { render, screen, waitFor } from '@testing-library/react';
import UserEvent from '@testing-library/user-event';
import EditComment from './index';

describe('EditComment Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <EditComment initialContent="" handleSubmit={() => {}} handleCancel={() => {}} />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    render(
      <EditComment initialContent="Initial Content" handleSubmit={() => {}} handleCancel={() => {}} />
    );

    expect(document.body).toMatchSnapshot();
  });

  it('renders an error as expected', async () => {
    const error = 'Test error';
    render(
      <EditComment
        initialContent="Initial Content"
        handleSubmit={() => new Promise((resolve, reject) => {
          throw new Error(error);
        })}
        handleCancel={() => {}}
      />
    );

    UserEvent.click(screen.getByText('Submit'));

    await waitFor(() => expect(screen.getByText(error)).toBeInTheDocument());
    expect(document.body).toMatchSnapshot();
  });
});