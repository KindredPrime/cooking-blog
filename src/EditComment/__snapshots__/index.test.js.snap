// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditComment Component renders an error as expected 1`] = `
<body>
  <div>
    <div
      class="EditComment"
    >
      <form
        class="EditComment"
      >
        <textarea
          id="content"
          name="content"
          required=""
        >
          Initial Content
        </textarea>
        <div
          class="EditComment__button-wrapper"
        >
          <button
            class="EditComment__submit"
            type="submit"
          >
            Submit
          </button>
          <button
            type="submit"
          >
            Cancel
          </button>
        </div>
      </form>
      <p
        class="APIError"
      >
        Test error
      </p>
    </div>
  </div>
</body>
`;

exports[`EditComment Component renders the UI as expected 1`] = `
<body>
  <div>
    <div
      class="EditComment"
    >
      <form
        class="EditComment"
      >
        <textarea
          id="content"
          name="content"
          required=""
        >
          Initial Content
        </textarea>
        <div
          class="EditComment__button-wrapper"
        >
          <button
            class="EditComment__submit"
            type="submit"
          >
            Submit
          </button>
          <button
            type="submit"
          >
            Cancel
          </button>
        </div>
      </form>
    </div>
  </div>
</body>
`;
